### PR TITLE
CMake changes for conda

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,10 +52,12 @@ endif()
 #general C/C++ build flags
 #/////////////////////////////////////////////////////////////////////////////////////
 
-set(CMAKE_BUILD_TYPE Debug)
-set(BUILD_STATIC_LIBS ON)
-set(BUILD_SHARED_LIBS OFF)
-add_definitions(-DENABLE_STATIC)
+set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build.")
```
Comment (isuruf) [inline CMakeLists.txt]: Making this a CACHE variable allows the user to override it

File: CMakeLists.txt [hunk 2]
```diff
@@ -247,22 +249,27 @@ endif()
 #compile as C++/static CRT on Visual Studio and as C99 on UNIX
 #//////////////////////////
 
+set(NCO_MSVC_USE_MT yes CACHE BOOL "Use MT flags when compiling in MSVC")
```
Comment (isuruf) [inline CMakeLists.txt]: Same here

File: CMakeLists.txt [hunk 3]
```diff
@@ -247,22 +249,27 @@ endif()
...
+
 if (MSVC)
   add_definitions( "/D_CRT_SECURE_NO_WARNINGS /D_CRT_NONSTDC_NO_DEPRECATE" )
-  message("-- Building with static runtime library")
-  set(CMAKE_CXX_FLAGS_DEBUG "/D_DEBUG /MTd /Zi /Ob0 /Od /RTC1")
```
Comment (isuruf) [inline CMakeLists.txt]: Using the default CMake values and modifying below allows the user to append to these from the command line.

**Comment by czender** [comment]
Note that the travis-ci failure is clang-only and is months old and was triggered when travis updated to ubuntu Trusty on 20170905, the link failure occurs in ncap2 not all because ncap2 links with g++ instead of clang. In other words this merge request appears good since no _new_ failures are reported.

**Comment by pedro-vicente** [comment]
@isuruf good addition...it seems what it does is to keep the current static, debug, and override if wanted on the command line, let's go ahead and merge

**Comment by ocefpaf** [comment]
Awesome! Thanks @isuruf for the patches and @czender and @pedro-vicente for the review.

**Pull request merged**